mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixtureModel.scala

```diff
@@ -83,5 +95,81 @@ class GaussianMixtureModel(
       p(i) /= pSum
     }
     p
-  }  
+  }
+}
+
+@Experimental
+object GaussianMixtureModel extends Loader[GaussianMixtureModel] {
+
+  private object SaveLoadV1_0 {
+
+    case class Data(weights: Double, mus: Vector, sigmas: Matrix)
+
+    def formatVersionV1_0 = "1.0"
+
+    def classNameV1_0 = "org.apache.spark.mllib.clustering.GaussianMixtureModel"
+
+    def save(
+        sc: SparkContext,
+        path: String,
+        weights: Array[Double],
+        gaussians: Array[MultivariateGaussian]): Unit = {
+
+      val sqlContext = new SQLContext(sc)
+      import sqlContext.implicits._
+
+      // Create JSON metadata.
+      val metadata = compact(render
+        (("class" -> classNameV1_0) ~ ("version" -> formatVersionV1_0) ~ ("k" -> weights.length)))
+      sc.parallelize(Seq(metadata), 1).saveAsTextFile(Loader.metadataPath(path))
+
+      // Create Parquet data.
+      val dataArray = Array.tabulate(weights.length){ i =>
+        Data(weights(i), gaussians(i).mu, gaussians(i).sigma)
+      }
+      val dataRDD: DataFrame = sc.parallelize(dataArray, 1).toDF()
+      dataRDD.saveAsParquetFile(Loader.dataPath(path))
+    }
+
+    def load(sc: SparkContext, path: String) : GaussianMixtureModel = {
+      val datapath = Loader.dataPath(path)
+      val sqlContext = new SQLContext(sc)
+      val dataRDD = sqlContext.parquetFile(datapath)
+      val numGaussians = dataRDD.count().toInt
+      val dataArray = dataRDD.select("weights", "mus", "sigmas").take(numGaussians)
```

Comment:
This is not efficient because it may trigger multiple passes to the parquet file. Let's call `collect()` directly.